# Analysis of Lifetime Value (LTV)

#####  Lifetime value (LTV) is the total worth to a business of a customer over the while period of their relationship. It's an important metric as it costs less to keep existings customers than to acquire new ones, so the idea of increasing the value of your existig customers is a great way to drive growth. 

In [31]:
#importing library
import sqlite3 as sql
import pandas as pd
import numpy as np

In [2]:
# creating connectiion and cursor
conn = sql.connect('churn.db')
cur = conn.cursor()

In [3]:
df = pd.read_sql("select * from cust_final", conn)
df

CustomerID  Gender SeniorCitizen Partner Dependents       State  \
0     3668-QPYBK    Male            No      No         No  California   
1     9237-HQITU  Female            No      No        Yes  California   
2     9305-CDSKC  Female            No      No        Yes  California   
3     7892-POOKP  Female            No     Yes        Yes  California   
4     0280-XJGEX    Male            No      No        Yes  California   
...          ...     ...           ...     ...        ...         ...   
7027  2569-WGERO  Female            No      No         No  California   
7028  6840-RESVB    Male            No     Yes        Yes  California   
7029  2234-XADUH  Female            No     Yes        Yes  California   
7030  4801-JZAZL  Female            No     Yes        Yes  California   
7031  3186-AJIEK    Male            No      No         No  California   

       Latitude   Longitude  ZipCode  Tenure  ... PhoneService  \
0     33.964131 -118.272783    90003       2  ...          Yes   
1     34.059281 -118.307420    90005       2  ...          Yes   
2     34.048013 -118.293953    90006       8  ...          Yes   
3     34.062125 -118.315709    90010      28  ...          Yes   
4     34.039224 -118.266293    90015      49  ...          Yes   
...         ...         ...      ...     ...  ...          ...   
7027  34.341737 -116.539416    92285      72  ...          Yes   
7028  34.667815 -117.536183    92301      24  ...          Yes   
7029  34.559882 -115.637164    92304      72  ...          Yes   
7030  34.167800 -116.864330    92305      11  ...           No   
7031  34.424926 -117.184503    92308      66  ...          Yes   

         MultipleLines InternetService       OnlineSecurity  \
0                   No             DSL                  Yes   
1                   No     Fiber optic                   No   
2                  Yes     Fiber optic                   No   
3                  Yes     Fiber optic                   No   
4                  Yes     Fiber optic                   No   
...                ...             ...                  ...   
7027                No              No  No internet service   
7028               Yes             DSL                  Yes   
7029               Yes     Fiber optic                   No   
7030  No phone service             DSL                  Yes   
7031                No     Fiber optic                  Yes   

             OnlineBackup     DeviceProtection          TechSupport  \
0                     Yes                   No                   No   
1                      No                   No                   No   
2                      No                  Yes                   No   
3                      No                  Yes                  Yes   
4                     Yes                  Yes                   No   
...                   ...                  ...                  ...   
7027  No internet service  No internet service  No internet service   
7028                   No                  Yes                  Yes   
7029                  Yes                  Yes                   No   
7030                   No                   No                   No   
7031                   No                  Yes                  Yes   

              StreamingTV      StreamingMovies Churn  
0                      No                   No   Yes  
1                      No                   No   Yes  
2                     Yes                  Yes   Yes  
3                     Yes                  Yes   Yes  
4                     Yes                  Yes   Yes  
...                   ...                  ...   ...  
7027  No internet service  No internet service    No  
7028                  Yes                  Yes    No  
7029                  Yes                  Yes    No  
7030                   No                   No    No  
7031                  Yes                  Yes    No  

[7032 rows x 25 columns]

# 1. Those customers who unsubscribed to service, what is their average LTV and how long do they stay alive in the service?

In [5]:
churn_df = pd.read_sql("select * from cust_final where churn = 'Yes';", conn)
churn_df

CustomerID  Gender SeniorCitizen Partner Dependents       State  \
0     3668-QPYBK    Male            No      No         No  California   
1     9237-HQITU  Female            No      No        Yes  California   
2     9305-CDSKC  Female            No      No        Yes  California   
3     7892-POOKP  Female            No     Yes        Yes  California   
4     0280-XJGEX    Male            No      No        Yes  California   
...          ...     ...           ...     ...        ...         ...   
1864  1699-HPSBG    Male            No      No         No  California   
1865  8775-CEBBJ  Female            No      No         No  California   
1866  6894-LFHLY    Male           Yes      No         No  California   
1867  0639-TSIQW  Female            No      No         No  California   
1868  8361-LTMKD    Male           Yes     Yes         No  California   

       Latitude   Longitude  ZipCode  Tenure  ... PhoneService MultipleLines  \
0     33.964131 -118.272783    90003       2  ...          Yes            No   
1     34.059281 -118.307420    90005       2  ...          Yes            No   
2     34.048013 -118.293953    90006       8  ...          Yes           Yes   
3     34.062125 -118.315709    90010      28  ...          Yes           Yes   
4     34.039224 -118.266293    90015      49  ...          Yes           Yes   
...         ...         ...      ...     ...  ...          ...           ...   
1864  33.745746 -116.514215    92264      12  ...          Yes            No   
1865  32.790282 -115.689559    92273       9  ...          Yes            No   
1866  34.264124 -114.717964    92280       1  ...          Yes           Yes   
1867  33.972293 -116.654195    92282      67  ...          Yes           Yes   
1868  34.550820 -117.122396    92307       4  ...          Yes           Yes   

     InternetService  OnlineSecurity  OnlineBackup DeviceProtection  \
0                DSL             Yes           Yes               No   
1        Fiber optic              No            No               No   
2        Fiber optic              No            No              Yes   
3        Fiber optic              No            No              Yes   
4        Fiber optic              No           Yes              Yes   
...              ...             ...           ...              ...   
1864             DSL              No            No               No   
1865             DSL              No            No               No   
1866     Fiber optic              No            No               No   
1867     Fiber optic             Yes           Yes              Yes   
1868     Fiber optic              No            No               No   

     TechSupport StreamingTV StreamingMovies Churn  
0             No          No              No   Yes  
1             No          No              No   Yes  
2             No         Yes             Yes   Yes  
3            Yes         Yes             Yes   Yes  
4             No         Yes             Yes   Yes  
...          ...         ...             ...   ...  
1864         Yes         Yes              No   Yes  
1865          No          No              No   Yes  
1866          No          No              No   Yes  
1867          No         Yes              No   Yes  
1868          No          No              No   Yes  

[1869 rows x 25 columns]

In [6]:
# finding distribution of total charges
churn_df['TotalCharges'].describe()

count    1869.000000
mean     1531.796094
std      1890.822994
min        18.850000
25%       134.500000
50%       703.550000
75%      2331.300000
max      8684.800000
Name: TotalCharges, dtype: float64

##### As we can see average total charges is very for 20% customers

In [7]:
# finding the 80th percentile
churn_df['TotalCharges'].quantile(0.8)


2840.4100000000003

In [11]:
#Divide the distribution into below and above 80
total_charges_below_80 = churn_df.query("TotalCharges<=2840.41")
total_charges_above_80 = churn_df.query("TotalCharges > 2840.41")

In [12]:
# show distribution of below 80 percentile
total_charges_below_80.TotalCharges.describe()

count    1495.000000
mean      711.265819
std       766.848197
min        18.850000
25%        85.025000
50%       371.650000
75%      1128.225000
max      2839.650000
Name: TotalCharges, dtype: float64

In [13]:
# show distribution of above 80 percentile
total_charges_above_80.TotalCharges.describe()

count     374.000000
mean     4811.723262
std      1436.724288
min      2841.550000
25%      3522.462500
50%      4571.100000
75%      5891.212500
max      8684.800000
Name: TotalCharges, dtype: float64

In [14]:
# show distribution of tenure of below 80 percentile
total_charges_below_80.Tenure.describe()

count    1495.000000
mean        9.935117
std        10.742349
min         1.000000
25%         1.000000
50%         6.000000
75%        15.000000
max        61.000000
Name: Tenure, dtype: float64

In [15]:
# show distribution of tenure of above 80 
total_charges_above_80.Tenure.describe()

count    374.000000
mean      50.133690
std       12.334841
min       27.000000
25%       40.000000
50%       49.500000
75%       60.000000
max       72.000000
Name: Tenure, dtype: float64

## The average LTV of those who unsubscribed to the service is $711 and tenure is approx. 10 months while for top 20%, average LTV is $4811 and tenure is near 50 months

# 2. When customers were subscribed, what services did they use? i.e. find top performing, worst performing services

In [19]:
### Phone service percentage 
# below 80 
total_charges_below_80.PhoneService.value_counts()/total_charges_below_80.shape[0]

Yes    0.892308
No     0.107692
Name: PhoneService, dtype: float64

In [20]:
#above 80
total_charges_above_80.PhoneService.value_counts()/total_charges_above_80.shape[0]

Yes    0.975936
No     0.024064
Name: PhoneService, dtype: float64

##### Only 2% of top 20% customers were not using phone services while in below 80% arond 11% were not using phone services i.e if you are not subscribed to phone service you are likely to be in below 80%

In [24]:
#below 80
total_charges_below_80_use_phone = total_charges_below_80.query("PhoneService == 'Yes'")
total_charges_below_80_use_phone.MultipleLines.value_counts()/total_charges_below_80_use_phone.shape[0]

No     0.590705
Yes    0.409295
Name: MultipleLines, dtype: float64

In [26]:
#above 80
total_charges_above_80_use_phone = total_charges_above_80.query("PhoneService == 'Yes'")
total_charges_above_80_use_phone.MultipleLines.value_counts()/total_charges_above_80_use_phone.shape[0]

Yes    0.832877
No     0.167123
Name: MultipleLines, dtype: float64

#### For the top 20% of LTV who used the phone service 84% of them used multiple lines, which is 2 times as much as the proportion of people in the low 80% LTV that is around 40%. Having multiple lines is going to get you closer to be in the top 20% of the LTV.

In [28]:
## INternet under 80
total_charges_below_80.InternetService.value_counts()/total_charges_below_80.InternetService.shape[0]

Fiber optic    0.640803
DSL            0.283612
No             0.075585
Name: InternetService, dtype: float64

In [29]:
#above 80
total_charges_above_80.InternetService.value_counts()/total_charges_above_80.InternetService.shape[0]

Fiber optic    0.906417
DSL            0.093583
Name: InternetService, dtype: float64

- In the high-LTV group, all the customers used the internet service and 90% of them used a fiber optic connection.
- On the other hand, among the low-LTV (80%) customers, 7% did not use the internet service and only 64% of them used a fiber optic connection. 
##### This suggests that having internet services, and specifically a fiber optic connection, can bring you closer to the high-LTV group.

In [34]:
## Other services below 80

total_charges_below_80_use_internet = total_charges_below_80.query("InternetService != 'No'")
sub_service_proportion_below_80 = np.array(
    [
        total_charges_below_80_use_internet.query("OnlineSecurity == 'Yes'").shape[0]/total_charges_below_80_use_internet.shape[0],
        total_charges_below_80_use_internet.query("OnlineBackup == 'Yes'").shape[0]/total_charges_below_80_use_internet.shape[0],
        total_charges_below_80_use_internet.query("DeviceProtection == 'Yes'").shape[0]/total_charges_below_80_use_internet.shape[0],
        total_charges_below_80_use_internet.query("TechSupport == 'Yes'").shape[0]/total_charges_below_80_use_internet.shape[0],
        total_charges_below_80_use_internet.query("StreamingTV == 'Yes'").shape[0]/total_charges_below_80_use_internet.shape[0],
        total_charges_below_80_use_internet.query("StreamingMovies == 'Yes'").shape[0]/total_charges_below_80_use_internet.shape[0],
])
sub_service_proportion_below_80

array([0.1316932 , 0.22720695, 0.23516643, 0.13603473, 0.37698987,
       0.3813314 ])

In [35]:
## Other services above 80

total_charges_above_80_use_internet = total_charges_above_80.query("InternetService != 'No'")
sub_service_proportion_above_80 = np.array(
    [
        total_charges_above_80_use_internet.query("OnlineSecurity == 'Yes'").shape[0]/total_charges_above_80_use_internet.shape[0],
        total_charges_above_80_use_internet.query("OnlineBackup == 'Yes'").shape[0]/total_charges_above_80_use_internet.shape[0],
        total_charges_above_80_use_internet.query("DeviceProtection == 'Yes'").shape[0]/total_charges_above_80_use_internet.shape[0],
        total_charges_above_80_use_internet.query("TechSupport == 'Yes'").shape[0]/total_charges_above_80_use_internet.shape[0],
        total_charges_above_80_use_internet.query("StreamingTV == 'Yes'").shape[0]/total_charges_above_80_use_internet.shape[0],
        total_charges_above_80_use_internet.query("StreamingMovies == 'Yes'").shape[0]/total_charges_above_80_use_internet.shape[0],
])
sub_service_proportion_above_80

array([0.30213904, 0.55882353, 0.58823529, 0.32620321, 0.78342246,
       0.77807487])

#### In top 20% around 80% customers used streamingTV, movies services while in below 80% on 40% used it.Online backup in low LTV is around 20% while in high LTV it is 56%. so in order to increase average LTV we can push customers to streaming tv and movies services.

# 3. For those customers who churned what is proportion of each kind of contracts?

In [38]:
# Under80 percentage
total_charges_below_80.Contract.value_counts()/total_charges_below_80.shape[0]

Month-to-month    0.959197
One year          0.036789
Two year          0.004013
Name: Contract, dtype: float64

In [39]:
total_charges_above_80.Contract.value_counts()/total_charges_above_80.shape[0]

Month-to-month    0.590909
One year          0.296791
Two year          0.112299
Name: Contract, dtype: float64

## 4. For those who did not churn what is the proportion of each kind of contract?

In [40]:
current_cust = pd.read_sql("select * from cust_final where churn == 'No';", conn)
current_cust

CustomerID  Gender SeniorCitizen Partner Dependents       State  \
0     7590-VHVEG  Female            No     Yes         No  California   
1     5575-GNVDE    Male            No      No         No  California   
2     7795-CFOCW    Male            No      No         No  California   
3     1452-KIOVK    Male            No      No        Yes  California   
4     6713-OKOMC  Female            No      No         No  California   
...          ...     ...           ...     ...        ...         ...   
5158  2569-WGERO  Female            No      No         No  California   
5159  6840-RESVB    Male            No     Yes        Yes  California   
5160  2234-XADUH  Female            No     Yes        Yes  California   
5161  4801-JZAZL  Female            No     Yes        Yes  California   
5162  3186-AJIEK    Male            No      No         No  California   

       Latitude   Longitude  ZipCode  Tenure  ... PhoneService  \
0     33.973616 -118.249020    90001       1  ...           No   
1     33.949255 -118.246978    90002      34  ...          Yes   
2     34.076259 -118.310715    90004      45  ...           No   
3     34.027337 -118.285150    90007      22  ...          Yes   
4     34.008293 -118.346766    90008      10  ...           No   
...         ...         ...      ...     ...  ...          ...   
5158  34.341737 -116.539416    92285      72  ...          Yes   
5159  34.667815 -117.536183    92301      24  ...          Yes   
5160  34.559882 -115.637164    92304      72  ...          Yes   
5161  34.167800 -116.864330    92305      11  ...           No   
5162  34.424926 -117.184503    92308      66  ...          Yes   

         MultipleLines InternetService       OnlineSecurity  \
0     No phone service             DSL                   No   
1                   No             DSL                  Yes   
2     No phone service             DSL                  Yes   
3                  Yes     Fiber optic                   No   
4     No phone service             DSL                  Yes   
...                ...             ...                  ...   
5158                No              No  No internet service   
5159               Yes             DSL                  Yes   
5160               Yes     Fiber optic                   No   
5161  No phone service             DSL                  Yes   
5162                No     Fiber optic                  Yes   

             OnlineBackup     DeviceProtection          TechSupport  \
0                     Yes                   No                   No   
1                      No                  Yes                   No   
2                      No                  Yes                  Yes   
3                     Yes                   No                   No   
4                      No                   No                   No   
...                   ...                  ...                  ...   
5158  No internet service  No internet service  No internet service   
5159                   No                  Yes                  Yes   
5160                  Yes                  Yes                   No   
5161                   No                   No                   No   
5162                   No                  Yes                  Yes   

              StreamingTV      StreamingMovies Churn  
0                      No                   No    No  
1                      No                   No    No  
2                      No                   No    No  
3                     Yes                   No    No  
4                      No                   No    No  
...                   ...                  ...   ...  
5158  No internet service  No internet service    No  
5159                  Yes                  Yes    No  
5160                  Yes                  Yes    No  
5161                   No                   No    No  
5162                  Yes                  Yes    No  

[5163 rows x 25 columns]

In [41]:
current_cust.TotalCharges.describe()

count    5163.000000
mean     2555.344141
std      2329.456984
min        18.800000
25%       577.825000
50%      1683.600000
75%      4264.125000
max      8672.450000
Name: TotalCharges, dtype: float64

In [43]:
#finding 80th percentile
current_cust.TotalCharges.quantile(0.8)

4895.85

In [44]:
#dividing the data into above and below 80th percentile
current_cust_below_80 = current_cust.query("TotalCharges <= 4895.85")
current_cust_above_80 = current_cust.query("TotalCharges > 4895.85")

In [45]:
#under 80 contract type percentage
current_cust_below_80.Contract.value_counts()/current_cust_below_80.shape[0]

Month-to-month    0.500726
Two year          0.253269
One year          0.246005
Name: Contract, dtype: float64

In [46]:
#above 80 contract type percentage
current_cust_above_80.Contract.value_counts()/current_cust_above_80.shape[0]

Two year          0.572120
One year          0.280736
Month-to-month    0.147144
Name: Contract, dtype: float64

- If we want to stop Churning we would like to sell contracts to people for at least one year, and much better for 2 years (Conjecture).
- We have 57% of the people that stayed in the company in the top 20% LTV with 2 year contracts, and only 11% of the people in the top 20% had a 2 year contract and left the company.
- 95% of the people that left the company and are in the low 80% have a month-to-month contract, in contrast to only 50% of the people in the low 80% that stayed in the company. We can say that having a month-to-month contract is one of the reasons that people are leaving the company (conjecture).
